main.py

```diff
@@ -116,29 +116,19 @@ def process(data, factors):
     return average
 
 def choose_direction(data, average):
-    user_surfaces = []
-    for direction_name, surface_arrays in data.items():
-        user_surfaces.append((direction_name, len(surface_arrays.keys())))
-    #get the number of surfaces stored in average as keys
-    user_surfaces = sorted(user_surfaces, key=lambda x: x[1], reverse=True)
-    chosen_surface_num = user_surfaces[0][1]
-    average_surface_num = len(average[next(iter(average))].keys())
-    #I am not sure if this is really necessary to check if they are equal
-    if average_surface_num == chosen_surface_num:
-        return user_surfaces[0][0]
-    
+    data_direction_surfaces = [(direction, data_surfaces.keys()) for direction, data_surfaces in data.items()]
+    average_surfaces = [average_surface for average_surface in average[next(iter(average))].keys()]
+    for direction, data_surfaces in data_direction_surfaces:
+        if all(average_surface in data_surfaces for average_surface in average_surfaces):
+            return direction  
            
 if __name__ == "__main__":
```

Comment:
@harrywhoforks Have you tried to run and test it before committing?

Reply:
yaya. just not sure why there is a typo...lol

Reply:
@harrywhoforks You should test the codes before committing bro

Reply:
@harrywhoforks have you tested it?